### PR TITLE
More CI_Form_validation unit tests

### PR DESCRIPTION
**Pull request by djwoods**
Currently tests for set_select, set_radio, and set_checkbox are failing. I've added an inline comment briefly describing the cause of the failure. Please let me know if the test case should be failing as designed (i.e. form_validation is working as expected).

Signed off by:
djwoods <d.woods92@gmail.com>

### CODE REVIEW

**Comment by narfbg** [comment]
Yes, it is working as expected.

**Comment by djwoods** [comment]
Ok. I've updated the tests accordingly.
Coverage is now at 75% for CI_Form_Validation. I think that's the highest I can get it

**Comment by narfbg** [comment]
Now these brackets that you've added are breaking it ...

**Comment by djwoods** [comment]
Care to elaborate? 
It's working on my machine - all tests are passing.

**Comment by narfbg** [comment]
https://travis-ci.org/bcit-ci/CodeIgniter/builds/56447633

The red X sign in front of your last commit points to that link. It fails on PHP 5.3 and 5.2.

**Comment by djwoods** [comment]
Thanks.
Issue present in the initial commit. It's now fixed

File: tests/codeigniter/libraries/Form_validation_test.php [hunk 1]
```diff
@@ -323,6 +323,254 @@ public function test_set_message()
...
+
+	public function test_error_array()
+	{
+		$this->form_validation->reset_validation();
+		$error_message = 'What a terrible error!';
+		$this->form_validation->set_message('required', $error_message);
+		$this->form_validation->set_rules('foo', 'label', 'required');
+		$_POST = array('foo' => '');
+		$this->form_validation->run();
+		$err_arr = $this->form_validation->error_array();
```
Comment (narfbg) [inline tests/codeigniter/libraries/Form_validation_test.php]: Please don't use such variable names ... $errors, $array, $error_array - each one is better than $err_arr.

File: tests/codeigniter/libraries/Form_validation_test.php [hunk 2]
```diff
@@ -313,7 +313,7 @@ public function test_set_message()
 		$this->form_validation->set_data($erroneous_data);
 		$this->form_validation->set_rules($rules);
 		$this->form_validation->run();
-		$this->assertEquals('<p>'.$err_message.'</p>', $this->form_validation->error('req_field'));
+		$this->assertEquals('<p>' . $err_message . '</p>', $this->form_validation->error('req_field'));
```
Comment (narfbg) [inline tests/codeigniter/libraries/Form_validation_test.php]: Please don't alter lines that are unrelated to your PR ... and we don't put spaces around concatenation.

File: tests/codeigniter/libraries/Form_validation_test.php [hunk 3]
```diff
@@ -342,4 +590,5 @@ public function run_rules($rules, $values)
 
 		return $this->form_validation->run();
 	}
+
```
Comment (narfbg) [inline tests/codeigniter/libraries/Form_validation_test.php]: When you add this, I remove it in the next commit ... don't. :)

**Comment by narfbg** [comment]
Other than these styling changes, it's all OK.

You're also inserting whitespace on a few lines that are supposed to be empty. Too nitpicky to point to these and I'll clean that up afterwards, but please look into your IDE's settings to see how to avoid that.

**Comment by djwoods** [comment]
I've updated the variable names.
I've also adjusted a few of my IDE's format settings. Hopefully it now aligns with your requirements...
